pystac/validation/stac_validator.py

```diff
@@ -159,9 +159,34 @@ def get_schema_from_uri(self, schema_uri: str) -> Tuple[Dict[str, Any], Any]:
 
         return schema, resolver
 
-    def _validate_from_uri(self, stac_dict: Dict[str, Any], schema_uri: str) -> None:
-        schema, resolver = self.get_schema_from_uri(schema_uri)
-        jsonschema.validate(instance=stac_dict, schema=schema, resolver=resolver)
+    def _validate_from_uri(
+        self,
+        stac_dict: Dict[str, Any],
+        stac_object_type: STACObjectType,
+        schema_uri: str,
+        extension_id: Optional[str] = None,
+        href: Optional[str] = None,
+    ) -> None:
+        try:
+            schema, resolver = self.get_schema_from_uri(schema_uri)
+            cls = jsonschema.validators.validator_for(schema)
+            cls.check_schema(schema)
+            validator = cls(schema, resolver=resolver)
+            errors = list(validator.iter_errors(stac_dict))
+        except Exception as e:
+            logger.error(f"Exception while validating {stac_object_type} href: {href}")
+            logger.exception(e)
+            raise
+        if errors:
+            msg = self._get_error_message(
+                schema_uri,
+                stac_object_type,
+                extension_id,
+                href,
+                stac_dict.get("id", None),
+            )
+            best = jsonschema.exceptions.best_match(errors)
+            raise pystac.STACValidationError(msg, source=errors) from best
```

Comment:
I like this -- keeps the original behavior, but augments the exception w/ the full suite of errors. 💡

Reply:
🐢 small improvements